.authentication.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+. "$(dirname "${BASH_SOURCE[0]}")/.common.sh"
```

Comment:
This whole file feels incorrect. We need to have proper support for authentication interactions, but not like this.

Also the name `basic_authentication` is very confusing as the word "basic" is kind of a keyword.